src/libstore/binary-cache-store.cc

```diff
@@ -308,16 +308,17 @@ void BinaryCacheStore::addToStore(const ValidPathInfo & info, Source & narSource
 }
 
 StorePath BinaryCacheStore::addToStoreFromDump(Source & dump, const string & name,
-    FileIngestionMethod method, HashType hashAlgo, RepairFlag repair)
+    FileIngestionMethod method, HashType hashAlgo, RepairFlag repair, StorePathSet references)
```

Comment:
```suggestion
    FileIngestionMethod method, HashType hashAlgo, RepairFlag repair, const StorePathSet & references)
```

and everywhere else.